backend/src/routes/aggregators/applicationsOCP.ts

```diff
@@ -125,8 +168,68 @@ export async function getOCPApps(applicationCache: ApplicationCacheType, argAppS
       },
     } as unknown as IResource)
   })
-  applicationCache['localOCPApps'] = generateTransforms(localOCPApps)
-  applicationCache['remoteOCPApps'] = generateTransforms(remoteOCPApps, true)
+  if (mode === MODE.ExcludeSystemApps) {
+    applicationCache['localOCPApps'] = generateTransforms(localOCPApps)
+    applicationCache['remoteOCPApps'] = generateTransforms(remoteOCPApps, true)
+  } else if (mode === MODE.OnlySystemApps) {
+    // if we just got remote clusters this time, don't touch localSysApps
+    if (localOCPApps.length) {
+      applicationCache['localSysApps'] = generateTransforms(localOCPApps)
+    }
+    // fill in remote system apps
+    fillRemoteSystemCache(applicationCache, remoteOCPApps, clusterNameChunk)
+  }
+}
+
+function getNextClusterNameChuck(applicationCache: ApplicationCacheType): string[] {
+  // if no cluster name chucks left, create a new array of chunks
+  if (clusterNameChunks.length === 0) {
+    const clusterMap = getClusterMap()
+    const clusterNames = Object.keys(clusterMap)
+    if (clusterNames.length > 0) {
+      const chunks = clusterNames.reduce((chunks: string[][], chunkName, index) => {
+        const cindex = Math.floor(index / REMOTE_CLUSTER_CHUNKS)
+        chunks[cindex] = ([] as string[]).concat(chunks[cindex] || [], chunkName)
```

Comment:
Two suggestions here:

1. Variable naming. A chunk appears to be an array of cluster names that itself does not have a name. I think a better name for `chunkName` would be `clusterName`.
2. This line is overly convoluted. Would one of the following work? They are concise and avoid the TypeScript cast.

```suggestion
        chunks[cindex] = [...(chunks[cindex] ?? []), clusterName]
        chunks[cindex] = (chunks[cindex] ?? []).concat(clusterName)
```